tests/src/item.test.ts

```diff
@@ -0,0 +1,398 @@
+    const itemClassIndex = 0;
+
+    const createItemClassMintIxs = await createMintNFTInstructions(
+      provider,
+      itemClassMintKeypair.publicKey
+    );
+
+    const itemClassData = {
+      settings: {
+        freeBuild: null,
+        childrenMustBeEditions: null,
+        builderMustBeHolder: null,
+        updatePermissiveness: [
+          {
+            permissivenessType: { tokenHolder: true },
+            inherited: { notInherited: true },
+          },
+        ],
+        buildPermissiveness: [
+          {
+            permissivenessType: { tokenHolder: true },
+            inherited: { notInherited: true },
+          },
+        ],
+        stakingWarmUpDuration: null,
+        stakingCooldownDuration: null,
+        stakingPermissiveness: null,
+        unstakingPermissiveness: null,
+        childUpdatePropagationPermissiveness: [
+          {
+            childUpdatePropagationPermissivenessType: { usages: true },
+            inherited: { notInherited: true },
+          },
+          {
+            childUpdatePropagationPermissivenessType: { components: true },
+            inherited: { notInherited: true },
+          },
+          {
+            childUpdatePropagationPermissivenessType: {
+              updatePermissiveness: true,
+            },
+            inherited: { notInherited: true },
+          },
+          {
+            childUpdatePropagationPermissivenessType: {
+              buildPermissiveness: true,
+            },
+            inherited: { notInherited: true },
+          },
+          {
+            childUpdatePropagationPermissivenessType: {
+              stakingPermissiveness: true,
+            },
+            inherited: { notInherited: true },
+          },
+          {
+            childUpdatePropagationPermissivenessType: {
+              freeBuildPermissiveness: true,
+            },
+            inherited: { notInherited: true },
+          },
+        ],
+      },
+      config: {
+        usageRoot: null,
+        usageStateRoot: null,
+        componentRoot: null,
+        usages: null,
+        components: null,
+      },
+    };
+
+    args = {
+      classIndex: new anchor.BN(itemClassIndex),
+      parentClassIndex: null,
+      space: new anchor.BN(300),
+      desiredNamespaceArraySize: 1,
+      updatePermissivenessToUse: { tokenHolder: true },
+      storeMint: false,
+      storeMetadataFields: false,
+      itemClassData,
+      parentOfParentClassIndex: null,
+    } as Instructions.Item.CreateItemClassArgs;
+
+    accounts = {
+      itemMint: itemClassMintKeypair.publicKey,
+      parent: null,
+      parentMint: null,
+      parentOfParentClassMint: null,
+      metadataUpdateAuthority: walletKeypair.publicKey,
+      parentUpdateAuthority: null,
+    } as Instructions.Item.CreateItemClassAccounts;
+
+    additionalArgs = {} as Instructions.Item.CreateItemClassAdditionalArgs;
+
+    const createItemClassIxs = await itemProgram.instruction.createItemClass(
```

Comment:
I normally like the chaining of instructions, makes things faster but I think in this case, if we broke them all up into seperate transactions we could setup and run more tests a lot quicker. This entire chain of instructions isnt very malleable to be able to try out different configurations, soft opinion on this one though

Reply:
Yes please. At least separate them into their own test case in the same describe block for the whole item flow. Otherwise it's a bit of a pain in the ass to see what part of this flow is failing from a glance. 

I'd rather have individual tests that use the item program send calls and not just the instruction calls directly. At least as the primary first tests. Combining the instructions would be a nice thing to test as well, but I wouldn't consider them as critical right now to get in place.

Reply:
We can split mint instruction and item instruction to individual transaction.
But we cannot make it as an individual test case because all previous steps are required for system-wide tests.

e.g.
`complete_item_escrow_build_phase` -> `start_item_escrow_build_phase` -> `create_item_escrow` -> `create_item_class`

If we make these steps as individual test cases, we cannot get the result such as item_class public key and/or item_escrow public key in the previous test case.

Reply:
Its not that they should be individual test cases its that they should be resuable so we can tweak the parameters and try many different combinations

Reply:
They can be individual tests as long as you don't make them run in parallel.

Reply:
If we arent running in parallel, whats the point of switching from Mocha? I think Mocha can also run parallel down the line, and it would be less work to not switch

Reply:
@jackrieck brings up a good point. I initially thought it would be ok to make the switch, but it now feels like it's going more against the grain to change to jest if we're not going to see more immediate benefits. It think we should drop the change to jest for now and we can alway re-evaluate it if we're having issues with using mocha.

Reply:
@jackrieck In this case, tests don't need to be run in parallel. But in future cases, there will be a lot of tests.
My previous project, there are 20+ test files in program/tests directory and every test file has 10 ~ 20 test cases.
If we use mocha in that case, testing will take 5 ~ 10 mins or more. But in jest, it only takes 15 ~ 40 secs.
And on the other hand, what do you think the difference between jest and mocha in usage? The only different think is the difference of expect and assert.
I already kept mocha in program tests and in system tests I prefer to use jest. I also want to update mocha to jest in program tests but for some reasons (as you mentioned before, such as unfamiliar) I kept the original one.
Of course, it's not for quarrel and I keep an open mind to use mocha or jest. :)

Reply:
The original purpose of using jest was to reduce the test run time in local laptop and github actions.
I agree with what @long-banana said. We don't need to spend a lot of energy here.

Reply:
I think for me it doesnt matter too much, in the end all tests like you mentioned before should be run with the same framework, if you are willing to update so that all tests including program use jest or mocha then I think thats the best course of action here. How does that sound?

Reply:
FWIW i think mocha does support parallel testing, but I havent ever tried it: https://mochajs.org/#parallel-tests